controllers/gateway/apirule_controller.go

```diff
@@ -340,3 +334,17 @@ func (r *APIRuleReconciler) getLatestApiRule(ctx context.Context, api *gatewayv1
 
 	return apiRule, nil
 }
+
+func (r *APIRuleReconciler) isApiRuleConvertedFromV1beta2(ctx context.Context, req ctrl.Request) bool {
+	convertedApiRule := &gatewayv1beta1.APIRule{}
+	err := r.Client.Get(ctx, req.NamespacedName, convertedApiRule)
```

Comment:
@werdes72 could we not request the CR twice and spare extra API server `get` request. It is requested again later in `Reconcile()`.